test/middleware/authenticate.redirect.test.js

```diff
@@ -42,6 +42,54 @@ describe('middleware/authenticate', function() {
       expect(response.getHeader('Content-Length')).to.equal('0');
     });
   });
+
+  describe('redirect with session', function() {
+    function Strategy() {
+    }
+    Strategy.prototype.authenticate = function(req, options) {
+      var user = { id: '1', username: 'idurotola' };
```

Comment:
We should start using `const` and `let` instead of `var` because package.json has the Node version at >=6.